### ISSUE TITLE
"MKL_VML" is undefined

### ISSUE THREAD

**Issue by klonuo** [pydata/numexpr#152]
Hi,

installing numexpr 2.4 on Ubuntu reports this error:

```
mkl_info:
  FOUND:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/composer_xe_2015/mkl/lib/intel64']
    define_macros = [('SCIPY_MKL_H', None)]
    include_dirs = ['/opt/intel/composer_xe_2015/mkl/include']

Warning: Assuming default configuration (numexpr/tests/{setup_tests,setup}.py was not found)Appending numexpr.tests configuration to numexpr
Ignoring attempt to set 'name' (from 'numexpr' to 'numexpr.tests')
/bin/sh: 1: svnversion: not found
running build
running config_cc
unifing config_cc, config, build_clib, build_ext, build commands --compiler options
running config_fc
unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
running build_src
build_src
building py_modules sources
creating build
creating build/src.linux-x86_64-2.7
creating build/src.linux-x86_64-2.7/numexpr
building extension "numexpr.interpreter" sources
build_src: building npy-pkg config files
running build_py
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/numexpr
copying numexpr/necompiler.py -> build/lib.linux-x86_64-2.7/numexpr
copying numexpr/utils.py -> build/lib.linux-x86_64-2.7/numexpr
copying numexpr/version.py -> build/lib.linux-x86_64-2.7/numexpr
copying numexpr/expressions.py -> build/lib.linux-x86_64-2.7/numexpr
copying numexpr/cpuinfo.py -> build/lib.linux-x86_64-2.7/numexpr
copying numexpr/__init__.py -> build/lib.linux-x86_64-2.7/numexpr
copying build/src.linux-x86_64-2.7/numexpr/__config__.py -> build/lib.linux-x86_64-2.7/numexpr
creating build/lib.linux-x86_64-2.7/numexpr/tests
copying ./numexpr/tests/__init__.py -> build/lib.linux-x86_64-2.7/numexpr/tests
copying ./numexpr/tests/test_numexpr.py -> build/lib.linux-x86_64-2.7/numexpr/tests
running build_ext
Found executable /opt/intel/composer_xe_2015.0.090/bin/intel64/icc
Could not locate executable ecc
customize IntelEM64TCCompiler
customize IntelEM64TCCompiler using build_ext
customize IntelEM64TCCompiler
customize IntelEM64TCCompiler using build_ext
building 'numexpr.interpreter' extension
compiling C++ sources
C compiler: icc -O3 -g -fPIC -fp-model strict -fomit-frame-pointer -openmp -xhost

creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/numexpr
compile options: '-DSCIPY_MKL_H -I/opt/intel/composer_xe_2015/mkl/include -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c'
extra options: '-funroll-all-loops'
icc: numexpr/numexpr_object.cpp
In file included from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarraytypes.h(1804),
                 from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarrayobject.h(17),
                 from numexpr/module.hpp(12),
                 from numexpr/numexpr_object.cpp(10):
/usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h(15): warning #1224: #warning directive: "Using deprecated NumPy API, disable it by "          "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
  #warning "Using deprecated NumPy API, disable it by " \
   ^

icc: numexpr/module.cpp
In file included from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarraytypes.h(1804),
                 from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarrayobject.h(17),
                 from numexpr/module.hpp(12),
                 from numexpr/module.cpp(12):
/usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h(15): warning #1224: #warning directive: "Using deprecated NumPy API, disable it by "          "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
  #warning "Using deprecated NumPy API, disable it by " \
   ^

numexpr/module.cpp(301): error: identifier "MKL_VML" is undefined
      mkl_domain_set_num_threads(max_num_threads, MKL_VML);
                                                  ^

compilation aborted for numexpr/module.cpp (code 2)
In file included from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarraytypes.h(1804),
                 from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarrayobject.h(17),
                 from numexpr/module.hpp(12),
                 from numexpr/module.cpp(12):
/usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h(15): warning #1224: #warning directive: "Using deprecated NumPy API, disable it by "          "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
  #warning "Using deprecated NumPy API, disable it by " \
   ^

numexpr/module.cpp(301): error: identifier "MKL_VML" is undefined
      mkl_domain_set_num_threads(max_num_threads, MKL_VML);
                                                  ^

compilation aborted for numexpr/module.cpp (code 2)
error: Command "icc -O3 -g -fPIC -fp-model strict -fomit-frame-pointer -openmp -xhost -DSCIPY_MKL_H -I/opt/intel/composer_xe_2015/mkl/include -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c numexpr/module.cpp -o build/temp.linux-x86_64-2.7/numexpr/module.o -funroll-all-loops" failed with exit status 2
```

Any ideas?

**Comment by FrancescAlted** [comment]
Fixed in rev 224ac6f.  Closing.

**Comment by RoyiAvital** [comment]
By the way, if I install Anaconda's Numpy which comes with Intel MKL, can `numexpr` use that for VML?
